src/main.ts

```diff
@@ -196,45 +204,55 @@ _Updated at ${new Date().toLocaleString('en-US', { timeZone: 'America/Los_Angele
   }
 }
 
-async function asyncForEach<T>(
-  array: T[],
-  callback: (item: T, i: number, arr: T[]) => Promise<void>
-): Promise<void> {
-  for (let index = 0; index < array.length; index++) {
-    await callback(array[index], index, array);
-  }
-}
-
 async function run(): Promise<void> {
-  const argocd = await setupArgoCDCommand();
+  let argoInsecure = '';
+  if (INSECURE === 'true') {
+    process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+    argoInsecure = '--insecure';
+  }
   const apps = await getApps();
   core.info(`Found apps: ${apps.map(a => a.metadata.name).join(', ')}`);
 
-  const diffs: Diff[] = [];
-
-  await asyncForEach(apps, async app => {
-    const command = `app diff ${app.metadata.name} --local=${app.spec.source.path}`;
-    try {
-      core.info(`Running: argocd ${command}`);
-      // ArgoCD app diff will exit 1 if there is a diff, so always catch,
-      // and then consider it a success if there's a diff in stdout
-      // https://github.com/argoproj/argo-cd/issues/3588
-      await argocd(command);
-    } catch (e) {
-      const res = e as ExecResult;
-      core.info(`stdout: ${res.stdout}`);
-      core.info(`stderr: ${res.stderr}`);
-      if (res.stdout) {
-        diffs.push({ app, diff: res.stdout });
-      } else {
-        diffs.push({
-          app,
-          diff: '',
-          error: e
-        });
-      }
-    }
+  const argocd = await setupArgoCDCommand(argoInsecure);
+
+  const limit = pLimit(20);
+
+  let diffs: Diff[] = [];
+  const input: Promise<void>[] = [];
+  apps.forEach(app => {
+    input.push(
+      limit(async () => {
+        const command = `app diff ${app.metadata.name} --revision=${github.context.payload.pull_request?.head?.sha}`;
+        try {
+          core.info(`Running: argocd ${command}`);
+          // ArgoCD app diff will exit 1 if there is a diff, so always catch,
+          // and then consider it a success if there's a diff in stdout
+          // https://github.com/argoproj/argo-cd/issues/3588
+          await argocd(command);
+        } catch (e) {
+          const res = e as ExecResult;
+          core.info(`stdout: ${res.stdout}`);
+          core.info(`stderr: ${res.stderr}`);
+          if (res.stdout) {
+            diffs.push({ app, diff: res.stdout });
+          } else {
+            diffs.push({
+              app,
+              diff: '',
+              error: e
+            });
+          }
+        }
+      })
+    );
+  });
+
+  await Promise.all(input);
+
+  diffs = diffs.sort((a, b) => {
+    return a.app.metadata.name.localeCompare(b.app.metadata.name);
   });
```

Comment:
Could be `diffs.sort((a, b) => a.app.metadata.name.localeCompare(b.app.metadata.name))`